internal/audit/agent_log.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2024 Mia srl
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package audit
+
+import (
+	"context"
+
+	"github.com/rond-authz/rond/internal/utils"
+	"github.com/rond-authz/rond/logging"
+)
+
+type logAgent struct {
+	l     logging.Logger
+	cache AuditCache
+}
+
+func NewLogAgent(l logging.Logger) Agent {
+	return &logAgent{
+		l:     l,
+		cache: &SingleRecordCache{},
+	}
+}
+
+func (a *logAgent) Trace(_ context.Context, auditInput Audit) {
+	data := a.cache.Load()
+
+	auditData := auditInput.toPrint()
+	if data != nil {
+		auditData.applyDataFromPolicy(data)
+	}
+
+	a.l.
+		WithField("trail", utils.ToMap("audit", auditData)).
```

Comment:
I think it would be better to set the `audit` tag closer to the Audit struct, since it could bring to bug if in the future the tag will be changed. Maybe add a toMap method to the audit struct?
Or at least a constant value

Reply:
I've added a constant